src/kernels/composable_kernel/include/tensor_operation/threadwise_generic_tensor_slice_copy.hpp

```diff
@@ -70,7 +70,8 @@ struct ThreadwiseGenericTensorSliceCopy_v4r2
     }
 
     template <typename SrcData, typename DstData>
-    __device__ void Run(const SrcData* p_src, DstData* p_dst) const
+    __device__ void
+    Run(const SrcData* p_src, DstData* p_dst, SrcData zeroVal = static_cast<SrcData>(0.0)) const
```

Comment:
Please add an member variable ```mSrcOutOfBoundValue``` to ```ThreadwiseGenericTensorSliceCopy_v4r2```, and initialize the former to the value you need, during the construction of  the latter.

Also, I think ```zero``` is not an accurate description of the variable under the context of threadwise copy, it's really just out-of-bound value.

Reply:
We can initialize the zeroVal using a member variable in the BlockwiseGenericTensorSliceCopy_v4r2 or ThreadwiseGenericTensorSliceCopy_v4r2 object constructor.  But be aware that zeroVal depends the type of the operation used (for Add, it is 0.0; for Min, it is 1.0), and the Blockwise/Threadwise copying object is not supposed to know the type of the operation in advance. And one single Blockwise/Threadwise copying object could be used to transfer data for different types of operations (Is this possible?).  So set the zeroVal by the member function Run() is actually a flexible way.

Reply:
Another not-good point is that we don't know the type of mSrcOutOfBoundValue until .Run() is called, so we have to use float/double if using mSrcOutOfBoundValue

Reply:
How about use  "float mOutOfBoundValue" and initialize it in the constructor and also add a member method SetOutOfBoundValue() ?

Reply:
Ok. I'm convinced that using source "out-of-bound" value as a member variable copy operator is a bad idea. Conceptually, it's an attribute of "transformed tensor" itself, but since we only have "tensor descriptor" but not "tensor" class, there is really nowhere to hold this attribute.

So keep your current implementation as it is, but please rename ```zeroVal``` to ```src_out_of_bound_value```

Reply:
Check https://github.com/ROCmSoftwarePlatform/MIOpen/pull/322/commits/83346911dc3b542028ba0a6b4eba2888adb084ba